codegen/src/codegen.rs

```diff
@@ -623,8 +623,11 @@ fn create_function(
 
     // auth
     ts_request_builder.extend(quote! {
-        if let Some(token) = &operation_config.bearer_access_token {
-            req_builder = req_builder.bearer_auth(token);
+        if let Some(token_credential) = &operation_config.token_credential {
+            let token_response = token_credential
```

Comment:
This will call `get_token` on every call on an API operation. Are we expecting that token credentials are in charge of their own caching?

Reply:
I am expecting the TokenCredential to handle any caching. That is consistent with the guidance:

> ⛔️ DO NOT persist, cache, or reuse tokens returned from the token credential. This is CRITICAL as credentials generally have a short validity period and the token credential is responsible for refreshing these.
https://azure.github.io/azure-sdk/general_design.html#authentication